### ISSUE TITLE
WinAppDriver fails to create new session on Surface. Got 404 error.

### ISSUE THREAD

**Issue by denys-rwd** [Microsoft/WinAppDriver#39]
Hi,

I have installed WinAppDriver on a brand-new Surface.

If I run the driver with localhost and run a **remote** selenium client then **/session** fails on **404** error.
If I run the driver with ip address and run a **remote** selenium client then  **/session**  succeed.

Looks like driver checks IP Address from client request to what was passed to driver from command line. It is very strange behavior. 

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Not Found</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY>**<h2>Not Found</h2>
<hr><p>HTTP Error 404. The requested resource is not found.</p>**
</BODY></HTML>
```

![session](https://cloud.githubusercontent.com/assets/18639833/16262385/157398d2-3878-11e6-9d4d-d34e40d19993.PNG)

**Comment by denys-rwd** [comment]
Any update?

**Comment by alakshmi1030** [comment]
Just to clarify - you have installed WinAppDriver on a Surface and you are targeting that machine from a Remote selenium client. Could you specify some cURL commands you are using from the remote client to target your WinAppDriver server so that we can try and repro the issue?

**Comment by denys-rwd** [comment]
**Test** - Open app

`
DesiredCapabilities capabilities = new DesiredCapabilities();
capabilities.setCapability("app", "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
rwd =  new RemoteWebDriver(new URL("http://172.30.201.161:4723/wd"), capabilities);
`


**Issue**
If WinAppDriver runs without input parameters, it starts automatically with 127.0.0.1.
![image](https://cloud.githubusercontent.com/assets/18639833/17134633/0912d2c0-5334-11e6-84d0-2dc2f082c655.png)
In the case, the test from remote machine **FAILED**. 
In the case, the test from local (where WinAppDriver is running) machine **SUCCESS**. 

If WinAppDriver gets IP Address as a input parameter.
![image](https://cloud.githubusercontent.com/assets/18639833/17134715/7dec7a06-5334-11e6-9418-a774e5ce2e49.png)
In the case, the test from remote machine is **SUCCESS**.


**Question**
Why to provide IP Address? IMO, Server must get a PORT only.

**Comment by denys-rwd** [comment]
I closed it by mistake. It is still relevant.

**Comment by paulcam206** [comment]
By default we don't want to bind to external-facing network interfaces for security reasons. It's by design that we're not remotely accessible.

**Comment by denys-rwd** [comment]
Consider to provide a new setting option. The option allows to set security mode. By default, it will be secured (local).  User can easy change it to unsecured (external) without providing external-facing IP Address. 

In my case, external IP address may change after restart. I don't like to update batch file on each restart. The above option will avoid redundant maintenance of batch file.

**Comment by denys-rwd** [comment]
Another strange case. 
If WinAppDriver is configured with external IP address and you run a test from LOCAL machine (where WinAppDriver is running) then 
 
Test with connect string that contains 172.30.201.161 will SUCCESS
`new RemoteWebDriver(new URL("http://**172.30.201.161**:4723/wd"), capabilities);`

Test with connect string that contains 127.0.0.1 will FAIL
`new RemoteWebDriver(new URL("http://**127.0.0.1**:4723/wd"), capabilities);`

WDYT?

Thanks in advance

**Comment by NickAb** [comment]
I am not familiar with WinAppDriver binding behaviour, but it seems that it binds to provided IP address only.

Also, the fact that it requires admin privileges to bind to non localhost suggests that driver might be using http.sys. It seems that it is possible to specify IP address as `*`. In that case driver will listen both on localhost and external ip address of you host machine. I've tried it locally and from the same machine I was able to get response both to `http://localhost:5000/status` and `http://10.211.55.6:5000/status`.
